pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1108,13 +1190,14 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(ctx context.Context, clusterName string
 func (lbaas *LbaasV2) ensureSecurityGroup(clusterName string, apiService *v1.Service, nodes []*v1.Node, loadbalancer *loadbalancers.LoadBalancer) error {
 	// find node-security-group for service
 	var err error
-	if len(lbaas.opts.NodeSecurityGroupIDs) == 0 {
+	if len(lbaas.opts.NodeSecurityGroupIDs) == 0 && !lbaas.opts.UseOctavia {
```

Comment:
why octavia is in this if?

Reply:
When using Octavia, the `lbaas.opts.NodeSecurityGroupIDs` is not gonna to be used, so we skip the logic here.